catalog/dags/providers/provider_dag_factory.py

```diff
@@ -537,18 +563,49 @@ def _build_partitioned_ingest_workflows(
     return partitioned_workflows, total_ingestion_metrics
 
 
+@task
+def report_aggregate_reingestion_errors(
+    provider_conf: ProviderReingestionWorkflow, dag_run=None
+):
+    """
+    Report ingestion errors that occurred during a reingestion workflow in
+    a single, aggregate slack message. If no errors were encountered, skip.
+    """
+    # Get the list of failed `pull_data` tasks
+    media_type_name = _get_media_type_descriptor(provider_conf.media_types)
+    failed_pull_data_tasks = [
+        task
+        for task in dag_run.get_task_instances(state=State.FAILED)
+        if f"pull_{media_type_name}_data" in task.task_id
+    ]
+
+    if not failed_pull_data_tasks:
+        raise AirflowSkipException
+
+    message = (
+        f"Ingestion errors were encountered in {len(failed_pull_data_tasks)}"
+        f" ingestion days while running the `{provider_conf.dag_id}` DAG. See the following"
+        " logs for details:\n"
+    ) + "\n".join(
+        f"  - <{task.log_url}|{task.task_id}>" for task in failed_pull_data_tasks
+    )
```

Comment:
Just noting that occasionally that `failed_pull_data_tasks` list can be hundreds of entries long, if something goes _really_ sideways 😮 do you think we should truncate the message when sending an alert, like we truncate the number of changed links we show for our documentation preview GitHub bot perhaps?

Reply:
It looks like the worst case scenario is if every Flickr reingestion day fails which would be 128 logfiles listed 😅 I don't think having all those links would even be particularly useful!

I'll go ahead and limit this to the first 5, I think. The number of failed tasks is still reported.